libs/libc/pthread/pthread_atfork.c

```diff
@@ -22,21 +22,65 @@
  * Included Files
  ****************************************************************************/
 
+#include <nuttx/tls.h>
+#include <nuttx/lib/lib.h>
+
 #include <pthread.h>
+#include <errno.h>
+#include <stdlib.h>
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name:  pthread_atfork
+ *
+ * Description:
+ *    To register the methods that need to be executed when fork is called
+ *    by any thread in a process
+ *
+ * Input Parameters:
+ *   prepare - the method that is executed in the parent process before
+ *             fork() processing is started
+ *   parent  - the method that is executed in the parent process after fork()
+ *             processing completes
+ *   child   - the method that is executed in the child process after fork()
+ *             processing completes
+ *
+ * Returned Value:
+ *   On success, pthread_atfork() returns 0.
+ *   On error, pthread_atfork() returns -1.
```

Comment:
returns not -1, but error number

Reply:
fix this issue in https://github.com/apache/nuttx/pull/9903